src/ColorPicker/Halogen/Layout.purs

```diff
@@ -0,0 +1,459 @@
+module ColorPicker.Halogen.Layout
+  ( Layout(..)
+  , ChildLayout(..)
+  , PickerComponent(..)
+
+  , NumberComponentView
+  , TextComponentView
+  , DragComponentView
+  , ActionComponentView
+
+  , InputTextValue
+  , isValid
+  , isInvalid
+
+  , PositionUpdate
+  , Dynamic
+  , PreNumConf
+  , InputProps
+  , Classes
+  , ValueHistory
+  , mapValueHistory
+
+  , LazyColor
+  , RecordHSLA
+  , RecordHSVA
+  , RecordRGBA
+  , mkLazyColor
+
+
+  , componentHue
+  , componentSaturationHSL
+  , componentLightness
+  , componentSaturationHSV
+  , componentValue
+  , componentRed
+  , componentGreen
+  , componentBlue
+  , componentHEX
+  , componentDragHue
+  , componentDragSV
+  , componentPreview
+  , componentHistory
+  , componentSet
+  ) where
+
+import Prelude
+
+import CSS as CSS
+import Color (Color)
+import Color as Color
+import Control.MonadZero (guard)
+import DOM.Event.Types (FocusEvent, MouseEvent, TouchEvent)
+import Data.Array (head, take)
+import Data.Int (floor, toNumber)
+import Data.Lazy (Lazy, defer, force)
+import Data.Maybe (Maybe(..), maybe, maybe')
+import Data.String as String
+import Halogen as H
+import Halogen.HTML as HH
+import Halogen.HTML.CSS as HCSS
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Math (round)
+import NumberInput.Halogen.Component as Num
+import NumberInput.Range (Range(..))
+
+data Layout = Root (Array H.ClassName) (Array ChildLayout)
+
+data ChildLayout
+  = Group (Array H.ClassName) (Array ChildLayout)
+  | Component PickerComponent
+
+
+type ValueHistory a =  { old ∷ Array a, current ∷ a }
+
+type InputTextValue = { value ∷ String, isValid ∷ Boolean }
+
+type PositionUpdate = { x ∷ Number, y ∷ Number } → Dynamic Color
+
+type Dynamic s = LazyColor → s
```

Comment:
`Dynamic Number` is just a Number which needs Color to be Computed. but it's not that of usefull will use `Color -> x `